storybookjs/storybook
# Build: Add missing dependencies to manager, unify TS to ~4.9.3

Issue:

Possible alternative to https://github.com/storybookjs/storybook/pull/20018

## What I did

The error mentioned in https://github.com/storybookjs/storybook/pull/20018 seems to indicate that `generate-exports-file.ts` might be running too early, before the packages that it is processing have been compiled.  By adding the missing dependencies from `ui/manager/src/globals/types.ts` into the package.json, they should be compiled by NX before manager is, hopefully solving the race condition.

This also brings all of the packages in the monorepo up to typescript `~4.9.3`, since a few were still at 4.6.3, and seemingly were causing build errors when `satisfies` was encountered.

## How to test

This is tricky, since the error only occurs occasionally.  One approach might be to re-run the unit tests in github CI multiple times, to see if the intermittent error still occurs.
